contrib/endpoints/src/api_manager/check_security_rules.cc

```diff
@@ -14,92 +14,350 @@
 //
 ////////////////////////////////////////////////////////////////////////////////
 #include "contrib/endpoints/src/api_manager/check_security_rules.h"
+#include <iostream>
+#include <sstream>
+#include "contrib/endpoints/src/api_manager/auth/lib/json_util.h"
+#include "contrib/endpoints/src/api_manager/proto/security_rules.pb.h"
+#include "contrib/endpoints/src/api_manager/utils/marshalling.h"
 
-#include <string>
-
-#include "contrib/endpoints/include/api_manager/api_manager.h"
-#include "contrib/endpoints/include/api_manager/request.h"
-
+using ::google::api_manager::auth::GetProperty;
+using ::google::api_manager::auth::GetStringValue;
 using ::google::api_manager::utils::Status;
+using ::google::protobuf::Map;
+using ::google::protobuf::util::error::Code;
 
 namespace google {
 namespace api_manager {
-
 namespace {
 
-const char kFirebaseServerStaging[] =
-    "https://staging-firebaserules.sandbox.googleapis.com/";
+const char kFailedFirebaseReleaseFetch[] = "Failed to fetch Firebase Release";
+const char kFailedFirebaseTest[] = "Failed to execute Firebase Test";
+const char kInvalidResponse[] = "Invalid JSON response from Firebase Service";
+const char kTestSuccess[] = "SUCCESS";
+const char kHttpGetMethod[] = "GET";
+const char kHttpPostMethod[] = "POST";
+const char kHttpHeadMethod[] = "HEAD";
+const char kHttpOptionsMethod[] = "OPTIONS";
+const char kHttpDeleteMethod[] = "DELETE";
+const char kFirebaseCreateMethod[] = "create";
+const char kFirebaseGetMethod[] = "get";
+const char kFirebaseDeleteMethod[] = "delete";
+const char kFirebaseUpdateMethod[] = "update";
+const char kV1[] = "v1/";
+const char kTestQuery[] = ":test?alt=json";
+const char kProjects[] = "projects/";
+const char kReleases[] = "/releases/";
+const char kRulesetName[] = "rulesetName";
+const char kTestResults[] = "testResults";
+const char kState[] = "state";
+const char kToken[] = "token";
+const char kAuth[] = "auth";
+const char kRequest[] = "request";
+const char kContentType[] = "Content-Type";
+const char kApplication[] = "application/json";
+
+const std::string GetFirebaseServer(const context::RequestContext &context) {
+  return context.service_context()
+      ->config()
```

Comment:
make a GetFireBaseServer() function in Config class.  Later when service_config is ready, get it from there. Only need to change that function.